addons/html_builder/static/src/website_builder/website_mass_mailing/mailing_list_subscribe_option_plugin.js

```diff
@@ -0,0 +1,113 @@
+import { MailingListSubscribeOption } from "./mailing_list_subscribe_option";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+import { user } from "@web/core/user";
+import { _t } from "@web/core/l10n/translation";
+
+class MailingListSubscribeOptionPlugin extends Plugin {
+    static id = "mailingListSubscribeOption";
+    static dependencies = ["remove", "savePlugin"];
+
+    resources = {
+        builder_options: [
+            {
+                OptionComponent: MailingListSubscribeOption,
+                props: {
+                    fetchMailingLists: this.fetchMailingLists.bind(this),
+                },
+                selector: ".s_newsletter_list",
+                cleanForSave: (editingElement) => {
+                    this.cleanForSave(editingElement);
+                },
+            },
+            {
+                template: "html_builder.MailingListSubscribeFormOption",
+                selector: ".s_newsletter_subscribe_form",
+            },
+        ],
+        builder_actions: [
+            {
+                toggleThanksMessage: {
+                    apply: ({ editingElement }) => {
+                        this.setThanksMessageVisibility(editingElement, true);
+                    },
+                    clean: ({ editingElement }) => {
+                        this.setThanksMessageVisibility(editingElement, false);
+                    },
+                    isApplied: ({ editingElement }) =>
+                        editingElement
+                            .querySelector(".js_subscribed_wrap")
+                            .classList.contains("o_enable_preview"),
+                },
+            },
+        ],
+        on_add_element_handlers: this.onAddElement.bind(this),
+    };
+
+    setThanksMessageVisibility(editingElement, isVisible) {
+        const toSubscribeEl = editingElement.querySelector(".js_subscribe_wrap");
+        const thanksMessageEl = editingElement.querySelector(".js_subscribed_wrap");
+        thanksMessageEl.classList.toggle("o_enable_preview", isVisible);
+        thanksMessageEl.classList.toggle("o_disable_preview", !isVisible);
+        toSubscribeEl.classList.toggle("o_enable_preview", !isVisible);
+        toSubscribeEl.classList.toggle("o_disable_preview", isVisible);
+    }
+
+    async onAddElement({ elementToAdd }) {
+        this.services.ui.block();
+        await this.fetchMailingLists();
+        if (this.mailingLists.length) {
+            elementToAdd.dataset.listId = this.mailingLists[0].id;
+        } else {
+            this.dependencies.remove.removeElement(elementToAdd);
```

Comment:
Be careful that if you do it here, the snippet will be removed even if we click on "confirm" -> should be done in the "cancel"